subsys/bluetooth/controller/ll_sw/ull_peripheral_iso.c

```
@@ -35,3 +80,200 @@ int ull_peripheral_iso_reset(void)
 {
 	return 0;
 }
+
+uint8_t ull_peripheral_iso_acquire(struct ll_conn *acl,
+				   struct pdu_data_llctrl_cis_req *req,
+				   uint16_t *cis_handle)
+{
+	struct ll_conn_iso_group *cig;
+	struct ll_conn_iso_stream *cis;
+
+	/* Get CIG by id */
+	cig = ll_conn_iso_group_get_by_id(req->cig_id);
+	if (!cig) {
+		/* CIG does not exist - create it */
+		cig = ll_conn_iso_group_acquire();
```

This should return an error if no CIG could be acquired.